piptools/repositories/pypi.py

```diff
@@ -7,22 +7,37 @@
 from contextlib import contextmanager
 from shutil import rmtree
 
-from pip.download import is_file_url, url_to_path
-from pip.index import PackageFinder
-from pip.req.req_set import RequirementSet
-from pip.wheel import Wheel
-from pip.utils.hashes import FAVORITE_HASH
+from .._compat import (
+    is_file_url,
+    url_to_path,
+    PackageFinder,
+    RequirementSet,
+    Wheel,
+    FAVORITE_HASH,
+    TemporaryDirectory,
+)
 
-from .._compat import TemporaryDirectory
 from ..cache import CACHE_DIR
 from ..exceptions import NoCandidateFound
 from ..utils import (fs_str, is_pinned_requirement, lookup_table,
                      make_install_requirement)
 from .base import BaseRepository
 
 
+try:
+    from pip._internal.operations.prepare import RequirementPreparer
+    from pip._internal.resolve import Resolver as PipResolver
+except ImportError:
+    pass
+
+try:
+    from pip._internal.cache import WheelCache
+except ImportError:
+    from pip.wheel import WheelCache
+
+
 class PyPIRepository(BaseRepository):
-    DEFAULT_INDEX_URL = 'https://pypi.python.org/simple'
+    DEFAULT_INDEX_URL = 'https://pypi.org/simple'
```

Comment:
The default index URL changes with the pip version, so I would make this match the one provided by pip itself if we stop vendoring pip.